src/tools/ts-generator/linksTSgenerator.cpp

```diff
@@ -166,52 +166,56 @@ void readIniProperties(std::vector<LinkTSgenerationParams>& linkList, fs::path t
     }
 }
 
-    bool readLinkPreproTimeSeries(LinkTSgenerationParams& link,
-                                  fs::path sourceAreaDir)
-    {
-        bool to_return = true;
-        const auto preproId = link.namesPair.first + "/" + link.namesPair.second;
-        link.prepro = std::make_unique<Data::PreproAvailability>(preproId, link.unitCount);
+fs::path makePreproFile(const fs::path& preproFilePath, const std::string& changingEnd)
+{
+    auto to_return = preproFilePath;
+    to_return += changingEnd + ".txt";
+    return to_return;
+}
 
-        auto preproFileRoot = sourceAreaDir / "prepro" / link.namesPair.second;
+bool readLinkPreproTimeSeries(LinkTSgenerationParams& link,
+                              fs::path sourceAreaDir)
+{
+    bool to_return = true;
+    const auto preproId = link.namesPair.first + "/" + link.namesPair.second;
+    link.prepro = std::make_unique<Data::PreproAvailability>(preproId, link.unitCount);
+
+    auto preproFileRoot = sourceAreaDir / "prepro" / link.namesPair.second;
+
+    // Testing files existence
```

Comment:
`// Testing files existence` : 
This file existence test was added because it used to exist in another form in this function, but disappeared by mistake.
So we reintroduce it under this new form.